### PR TITLE
[client/network] remove peer entry from `ephemeral_addresses`

### PR DESCRIPTION
**Pull request by melekes**
if there are no addresses associated with that peer

Refs #8466

### CODE REVIEW

**Comment by melekes** [comment]
**this is basically some small refactoring + removing entry if there are no addresses.**

**Review by dmitry-markin — APPROVED** [review]

File: client/network/src/discovery.rs [hunk 1]
```diff
@@ -539,6 +541,9 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 							for (addr, _error) in errors {
 								list.retain(|a| a != addr);
 							}
+							if list.is_empty() {
+								self.ephemeral_addresses.remove(&peer_id);
+							}
```
Comment (bkchr) [inline client/network/src/discovery.rs:546]: if let Entry::Occupied(entry) = self.ephemeral_addresses.entry(peer_id) {
 							for (addr, _error) in errors {
								entry.get_mut().retain(|a| a != addr);
							}
							if entry.get().is_empty() {
								entry.remove();
							}
}

Comment (bkchr) [reply client/network/src/discovery.rs:546]: This should replace the entire `if let Some` block

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2259829

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2259878

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-frame-examples-compile-to-wasm
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2259873

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2259863

**Comment by melekes** [comment]
bot rebase

**Comment by melekes** [comment]
bot merge

**Comment by melekes** [comment]
bot merge

**Pull request merged**